### PR TITLE
Add TO Client API for Service Categories Automation

### PR DESCRIPTION
**Pull request by dmohan001c**
## What does this PR (Pull Request) do?

- [x] This PR is not related to any Issue

## Which Traffic Control components are affected by this PR?

- CDN in a Box
- Traffic Control Client 
- Traffic Ops
- CI tests

## What is the best way to verify this PR?
Execute all the Integration tests and make sure the tests passed.

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_ops/testing/api/v4/servicecategories_test.go [hunk 1]
```diff
@@ -104,6 +112,34 @@ func CreateTestServiceCategories(t *testing.T) {
...
+		t.Fatalf("Expected 400 status code, got %v", reqInf.StatusCode)
+	}
+}
+
+func CreateTestServiceCategoriesInvalidName(t *testing.T) {
+	firstServiceCategory := tc.ServiceCategory{}
+	firstServiceCategory.Name = ""
+	resp, reqInf, err := TOSession.CreateServiceCategory(firstServiceCategory, client.RequestOptions{})
+	if err == nil {
+		t.Errorf("Expected service_category name already exists. but got: %v - alerts: %+v", err, resp.Alerts)
```
Comment (ocket8888) [inline traffic_ops/testing/api/v4/servicecategories_test.go:136]: I don't believe the expectation is that the service category name already exists in this test

File: traffic_ops/testing/api/v4/servicecategories_test.go [hunk 2]
```diff
@@ -233,3 +269,139 @@ func DeleteTestServiceCategories(t *testing.T) {
...
+	opts.QueryParameters = url.Values{}
+	opts.QueryParameters.Set("orderby", "id")
+	opts.QueryParameters.Set("limit", "1")
+	serviceCategoriesWithLimit, _, err := TOSession.GetServiceCategories(opts)
+	if err == nil {
+		if !reflect.DeepEqual(serviceCategories[:1], serviceCategoriesWithLimit.Response) {
+			t.Error("expected GET Service Categories with limit = 1 to return first result")
+		}
+	} else {
+		t.Error("Error in getting Service Categories by limit")
```
Comment (ocket8888) [inline traffic_ops/testing/api/v4/servicecategories_test.go:314]: This should print what the error actually was

File: traffic_ops/testing/api/v4/servicecategories_test.go [hunk 3]
```diff
@@ -233,3 +269,139 @@ func DeleteTestServiceCategories(t *testing.T) {
...
+	opts.QueryParameters.Set("orderby", "id")
+	opts.QueryParameters.Set("limit", "1")
+	opts.QueryParameters.Set("offset", "1")
+	serviceCategoriesWithOffset, _, err := TOSession.GetServiceCategories(opts)
+	if err == nil {
+		if !reflect.DeepEqual(serviceCategories[1:2], serviceCategoriesWithOffset.Response) {
+			t.Error("expected GET Service Categories with limit = 1, offset = 1 to return second result")
+		}
+	} else {
+		t.Error("Error in getting Service Categories by limit and offset")
```
Comment (ocket8888) [inline traffic_ops/testing/api/v4/servicecategories_test.go:327]: same as above RE: printing the error

File: traffic_ops/testing/api/v4/servicecategories_test.go [hunk 4]
```diff
@@ -233,3 +269,139 @@ func DeleteTestServiceCategories(t *testing.T) {
...
+	opts.QueryParameters.Set("orderby", "id")
+	opts.QueryParameters.Set("limit", "1")
+	opts.QueryParameters.Set("page", "2")
+	serviceCategoriesWithPage, _, err := TOSession.GetServiceCategories(opts)
+	if err == nil {
+		if !reflect.DeepEqual(serviceCategories[1:2], serviceCategoriesWithPage.Response) {
+			t.Error("expected GET Service Categories with limit = 1, page = 2 to return second result")
+		}
+	} else {
+		t.Error("Error in getting Service Categories by limit and page")
```
Comment (ocket8888) [inline traffic_ops/testing/api/v4/servicecategories_test.go:340]: same as above RE: printing the error

File: traffic_ops/testing/api/v4/servicecategories_test.go (hunk 1 above)
Comment (dmohan001c) [reply traffic_ops/testing/api/v4/servicecategories_test.go]: updated error message.

File: traffic_ops/testing/api/v4/servicecategories_test.go (hunk 2 above)
Comment (dmohan001c) [reply traffic_ops/testing/api/v4/servicecategories_test.go]: Added err in error print statement

File: traffic_ops/testing/api/v4/servicecategories_test.go (hunk 3 above)
Comment (dmohan001c) [reply traffic_ops/testing/api/v4/servicecategories_test.go]: Added err in error print statement

File: traffic_ops/testing/api/v4/servicecategories_test.go (hunk 4 above)
Comment (dmohan001c) [reply traffic_ops/testing/api/v4/servicecategories_test.go]: Added err in error print statement

**Review by ocket8888 — APPROVED** [review]

**Pull request merged**